lib/fs.js

```diff
@@ -548,17 +564,41 @@ function open(path, flags, mode, callback) {
   } else if (typeof mode === 'function') {
     callback = mode;
     mode = 0o666;
-  } else {
-    mode = parseFileMode(mode, 'mode', 0o666);
   }
-  const flagsNumber = stringToFlags(flags);
+  // Handle case where 2nd argument is options-bag
+  if (typeof flags === 'object' && flags) {
```

Comment:
If you're only using it in an if statement to check whether flags is an object and you don't need the value of flags itself, then using typeof flags === 'object' is sufficient:
```suggestion
  if (typeof flags === 'object') {
```

Reply:
That would crash if `flags` was `null`, as `typeof null === "object"`